openapi.yaml

```diff
@@ -0,0 +1,492 @@
+                $ref: "#/components/schemas/PassportResponse"
+        "400":
+          description: "Invalid data"
+
+
+  /person/{personId}/passport/{id}:
+    get:
+      tags:
+        - "PassportStore"
+      summary: "Finds person`s passport by ID"
+      description: "Gives person`s passport with current ID"
+      operationId: "getPassport"
+      parameters:
+        - name: "personId"
+          in: "path"
+          description: "Person`s ID to get"
+          required: true
+          schema:
+            type: "string"
+        - name: "id"
+          in: "path"
+          description: "ID of passport to get"
+          required: true
+          schema:
+            type: "string"
+        - name: "active"
+          in: "query"
+          description: "Status of passport to get"
+          schema:
+            type: "boolean"
+      responses:
+        "200":
+          description: "Successful operation"
+          content:
+            '*/*':
+              schema:
+                $ref: "#/components/schemas/PassportResponse"
+        "400":
+          description: "Invalid data"
+    put:
+      tags:
+        - "PassportStore"
+      summary: "Update current passport"
+      description: "Update person`s passport by ID"
+      operationId: "updatePassport"
+      parameters:
+        - name: "personId"
+          in: "path"
+          description: "Person`s ID to get"
+          required: true
+          schema:
+            type: "string"
+        - name: "id"
+          in: "path"
+          description: "ID of passport to update"
+          required: true
+          schema:
+            type: "string"
+      requestBody:
+        content:
+          'application/json':
+            schema:
+              $ref: '#/components/schemas/PassportRequest'
+      responses:
+        "200":
+          description: "Successful operation"
+          content:
+            '*/*':
+              schema:
+                $ref: "#/components/schemas/PassportResponse"
+        "400":
+          description: "Invalid data"
+    delete:
+      tags:
+        - "PassportStore"
+      summary: "Delete person`s passport with ID"
+      description: "Delete passport by ID"
+      operationId: "deletePassport"
+      parameters:
+        - name: "personId"
+          in: "path"
+          description: "Person`s ID to get"
+          required: true
+          schema:
+            type: "string"
+        - name: "id"
+          in: "path"
+          description: "Passport ID to delete"
+          required: true
+          schema:
+            type: "string"
+      responses:
+        "200":
+          description: "Successful operation"
+          content:
+            '*/*':
+              schema:
+                $ref: "#/components/schemas/PassportResponse"
+        "400":
+          description: "Invalid data"
+        "404":
+          description: "Passport not found"
+
+  /person/{personId}/passport/{id}/lostPassport:
+    patch:
```

Comment:
makes sense to use `POST` isntead, and describe a possible error when a passport is already lost

Reply:
fixed